File: Min Stack.java
class MinStack {
    private Stack<Integer> m_value = new Stack<Integer>();
    private Stack<Integer> m_min = new Stack<Integer>();
    public void push(int x) {
        m_value.push(x);
        if (m_min.empty() || x <= m_min.peek())
            m_min.push(x);
    }

    public void pop() {
        int val = m_value.pop();
        if (m_min.peek() == val)
            m_min.pop();
    }

    public int top() {
        return m_value.peek();
    }

    public int getMin() {
        return m_min.peek();
    }
}

File: Min Stack.cpp
class MinStack {
private:
    stack<int> m_value;
    stack<int> m_min;
public:
    void push(int x) {
        m_value.push(x);
        if (m_min.empty() || x <= m_min.top()) 
            m_min.push(x);
    }

    void pop() {
        int val = m_value.top();
        m_value.pop();
        if (m_min.top() == val)
            m_min.pop();
    }

    int top() {
        return m_value.top();
    }

    int getMin() {
        return m_min.top();
    }
};